src/Spice86.Core/Emulator/ReverseEngineer/MemoryBasedDataStructureWithBaseAddressProvider.cs

```diff
@@ -60,19 +50,7 @@ public abstract class MemoryBasedDataStructureWithBaseAddressProvider : MemoryBa
         SetUint8(BaseAddress, offset, value);
     }
 
-    public void SetZeroTerminatedString(int start, string value, int maxLength) {
-        if (value.Length + 1 > maxLength) {
-            throw new UnrecoverableException($"String {value} is more than {maxLength} cannot write it at offset {start}");
-        }
-
-        uint physicalStart = (uint)(BaseAddress + start);
-        int i = 0;
-        for (; i < value.Length; i++) {
-            char character = value[i];
-            byte charFirstByte = Encoding.ASCII.GetBytes(character.ToString())[0];
-            SetUint8(physicalStart, i, charFirstByte);
-        }
-
-        SetUint8(physicalStart, i, 0);
+    public void SetZeroTerminatedString(uint start, string value, int maxLength) {
+        MemoryUtils.SetZeroTerminatedString(this.Memory.Ram, BaseAddress + start, value, maxLength);
```

Comment:
You don't wamt to use Memory.Ram for this.
That creates a _copy_ of the entire memory.

I created a PR (https://github.com/OpenRakis/Spice86/pull/299) that allows you to do 
```suggestion
        Memory.SetZeroTerminatedString(BaseAddress + start, value, maxLength);
```

Reply:
Done, thank you ! :)